### DISCUSSION TITLE
What is the best strategy for reading a pdf w/multiple choice questions?

### DISCUSSION THREAD

**Discussion by metalaboratorio** [adrienjoly/npm-pdfreader#161] [Q&A]
Should I transform the entire text at once or use the rules to search for each question separately.
My first idea was to search for the questions - "1)" - and then each option -"a)"- in a different group, but it didn't work. 
[Page 6.pdf](https://github.com/user-attachments/files/17892455/Page.6.pdf)
My intention is to transform these questions in athe pdf file into HTML form. 
I think I'm doing something wrong, I'd appreciate it if someone could help me, tks.

**Comment by adrienjoly** [comment]
That's an interesting case, thanks for sharing the PDF!

The key is to design:
- how to recognize each question => how to detect the beginning of a question 
- how to recognize each part of the question: the question itself + its number + each possible answer 

In this case, it looks like each question is preceded by a number and a parenthesis => we could use that to recognize each question.

If I remember well, pdfreader's rule-based parser is not able to loop its logic over a document that repeats the same structural pattern more than once.

In that case, I see two possibilities:
- your parser could read all text entries from the file, looking for the beginning of a question and then redirect these text entries to pdfreader's rule-based processor in order to split it into sections.
- or you could write your own logic based on text entries to achieve the latter.

Good luck, and please share your results!

**Comment by adrienjoly** [comment]
Also, did you try asking ChatGPT for a solution, yet? (Either a high level solution, or code)
